### PR TITLE
Fix: dataframe.isna on scaler args

### LINKED ISSUE #54: [BUG] dataframe.isna fails on scaler arguments
### Describe the bug
`dataframe.isna` raises an exception when the input is a scaler.

### To Reproduce
```
import mars.dataframe as md
md.isna("foo")
```

### Expected behavior
returning `False`

### PR DESCRIPTION
**Pull request by UranusSeven**
## What do these changes do?
let `dataframe.isna` handle scalers.

## Related issue number
Resolves #54 

## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/dataframe/missing/checkna.py [hunk 1]
```diff
@@ -57,8 +60,13 @@ def __call__(self, df):
             self.output_types = [OutputType.dataframe]
         elif isinstance(df, SERIES_TYPE):
             self.output_types = [OutputType.series]
-        else:
+        elif isinstance(df, TENSOR_TYPE):
             self.output_types = [OutputType.tensor]
+        else:
+            # all the other types are treated as scalar, including pandas dataframe.
+            if df is None or df is md.NA or df is md.NaT or df is mt.nan:
```
[the remark this answers is not in the document]
Comment (UranusSeven) [reply mars/dataframe/missing/checkna.py:68]: > How about just use `pd.isna(df)`?

Using `pd.isna` means pandas dataframes are treated in the same way as mars dataframes. But it is not true for some other general functions like `unique`.

I think we'd better have a consistent way of dealing with pandas dataframes.

File: mars/dataframe/missing/checkna.py [hunk 2]
```diff
@@ -57,8 +61,13 @@ def __call__(self, df):
             self.output_types = [OutputType.dataframe]
         elif isinstance(df, SERIES_TYPE):
             self.output_types = [OutputType.series]
-        else:
+        elif isinstance(df, TENSOR_TYPE) or isinstance(df, INDEX_TYPE):
```
Comment (aresnow1) [inline mars/dataframe/missing/checkna.py:64]: It's better to return value directly in `isna` for those scalars, no need to create the operand.

File: mars/dataframe/missing/checkna.py [hunk 3]
```diff
@@ -98,14 +105,29 @@ def execute(cls, ctx, op: "DataFrameCheckNA"):
         try:
             pd.set_option("mode.use_inf_as_na", op.use_inf_as_na)
             if op.positive:
-                ctx[op.outputs[0].key] = in_data.isna()
+                ctx[op.outputs[0].key] = pd.isna(in_data)
```
Comment (aresnow1) [inline mars/dataframe/missing/checkna.py:108]: Better to handle GPU cases, I checked that cudf has no attribute 'isna'.

**Review by aresnow1 — DISMISSED** [review]
LGTM

File: mars/dataframe/missing/checkna.py [hunk 4]
```diff
@@ -57,7 +64,7 @@ def __call__(self, df):
             self.output_types = [OutputType.dataframe]
         elif isinstance(df, SERIES_TYPE):
             self.output_types = [OutputType.series]
-        else:
+        elif isinstance(df, TENSOR_TYPE) or isinstance(df, INDEX_TYPE):
```
Comment (qianduoduo0904) [inline mars/dataframe/missing/checkna.py:67]: For `else`, maybe add some assertion to ensure correctness.

File: mars/dataframe/missing/tests/test_missing.py [hunk 5]
```diff
@@ -222,3 +223,206 @@ def test_replace():
...
     assert r.chunks[0].op.stage is None
     assert r.chunks[0].op.limit is None
+
+
+@pytest.mark.parametrize("inf_as_na", [True, False])
+def test_isna(setup, inf_as_na):
+    from ..checkna import isna
+    from ....config import options
+
+    options.dataframe.mode.use_inf_as_na = inf_as_na
```
Comment (qianduoduo0904) [inline mars/dataframe/missing/tests/test_missing.py:233]: If we modify the options in test, we'd better set back the original one.

File: mars/dataframe/missing/tests/test_missing.py [hunk 6]
```diff
@@ -222,3 +223,206 @@ def test_replace():
...
+
+
+@pytest.mark.parametrize("inf_as_na", [True, False])
+def test_notna(setup, inf_as_na):
+    from ..checkna import notna
+    from ....config import options
+
+    options.dataframe.mode.use_inf_as_na = inf_as_na
+    # this option could be changed by mars execution.
+    pd.options.mode.use_inf_as_na = inf_as_na
```
Comment (qianduoduo0904) [inline mars/dataframe/missing/tests/test_missing.py:337]: Ditto.

**Review by qianduoduo0904 — APPROVED** [review]
LGTM

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**